### PR TITLE
Chore/refactor params

### PR DESCRIPTION
**Pull request by vigan-abd**
### Description:

Refactored all methods to use format `(params, cb = null)`

### Breaking changes:
- method signatures

### New features:
- [ ]

### Fixes:
- [ ]

### PR status:
- [ ] Version bumped
- [ ] Change-log updated
- [x] Tests added or updated
- [x] Documentation updated

### CODE REVIEW

**Review by avsek477 — APPROVED** [review]
If the default values taken for certain fields before introducing the params object (like the default symbol for most cases was `tBTCUSD`) should not be the same for the params object now, then lgtm.
Also since, it's a breaking change, better to upgrade the version?

File: package.json
Comment (tarcisiozf) [inline package.json:1]: We have a lot of breaking changes regarding params, I think we should change the major version here

File: lib/rest2.js [hunk 1]
```diff
@@ -270,7 +268,7 @@ class RESTv2 {
   /**
    * @param {object} data - data
    * @param {object|Function} transformer - model class or function
-   * @param {Function} [cb] - callback
+   * @param {Function} [cb] - legacy callback
    * @returns {object|object[]} finalData
```
Comment (tarcisiozf) [inline lib/rest2.js:272]: _cb returns a promise, the doc return type should be Promise too

File: lib/rest2.js [hunk 2]
```diff
@@ -1325,137 +1408,57 @@ class RESTv2 {
...
   /**
    * Cancel existing order using the cID
    *
-   * @param {number} cid - cid
-   * @param {string} date - Date of order YYYY-MM-DD
-   * @param {Method} cb - callback
+   * @param {object} params      - parameters
+   * @param {number} params.cid  - cid
+   * @param {string} params.date - Date of order YYYY-MM-DD
+   * @param {Method} cb          - legacy callback
```
Comment (tarcisiozf) [inline lib/rest2.js:1443]:
```suggestion
   * @param {Function} cb          - legacy callback
```

File: lib/rest2.js [hunk 3]
```diff
@@ -1750,11 +1769,11 @@ class RESTv2 {
    * @param {number} params.isPublic    - make Pulse message public
    * @param {string} params.isPin       - make Pulse message pinned
    * @param {string} params.attachments - Base64 encoded list of strings
-   * @param {Function} [cb]             - callback
+   * @param {Function} [cb]             - legacy callback
    * @returns {Promise} p
    * @see https://docs.bitfinex.com/reference#rest-auth-pulse-add
    */
-  addPulseComment (params, cb) {
+  addPulseComment (params, cb = null) {
```
Comment (tarcisiozf) [inline lib/rest2.js:1776]: Weird that this method calls addPulse with same params but the docs does not match, this one has params.parent and addPulse has params.title

File: lib/rest2.js [hunk 4]
```diff
@@ -1843,28 +1861,53 @@ class RESTv2 {
    * @param {string} [params.customerInfo.residBuildingNo] - optional, customer's residential building number/name
    * @param {string} params.customerInfo.fullName          - customer's full name
    * @param {string} params.customerInfo.email             - customer's email address
-   * @param {Function} [cb]                                - callback
+   * @param {object} [params.meta]                         - metadata
+   * @param {string} [params.meta.env]                     - merchant environment, allowed values: production, test,
+   *                                                         staging, development
+   * @param {string[]} [param.meta.tags]                   - invoice tags, max number of tags is 5 and max length of
```
Comment (tarcisiozf) [inline lib/rest2.js:1867]:
```suggestion
   * @param {string[]} [params.meta.tags]                   - invoice tags, max number of tags is 5 and max length of
```

File: package.json
Comment (vigan-abd) [reply package.json:1]: yep that's intended after we merge this and migration to node-fetch

File: lib/rest2.js (hunk 3 above)
Comment (vigan-abd) [reply lib/rest2.js:1776]: I think title is required only when parent is missing

**Pull request merged**